### DISCUSSION TITLE
How to Authenticate users and get user ID, On custom route.

### DISCUSSION THREAD

**Discussion by Johny-rescaf** [pocketbase/pocketbase#1793] [Q&A] [answered]
I am suing pocketbase as a framework and i am adding a custom route where i will perform payment. i have done all the logic already but i need a way to authenticate the user accessing the route and get the userID for futher processing.
Thanks for any help.

**Comment by ganigeorgiev** [comment, marked as answer]
You can find an OAuth2 client-side example in https://pocketbase.io/docs/authentication/#web-oauth2-integration.

I'm not sure that I understand your use case, but you can find the OAuth2 user data from the auth response under the `meta` prop (including the OAuth2 refresh/access tokens if you need to perform some other custom OAuth2 actions).

**Reply by Johny-rescaf** [reply]
Thanks for the reply. Actually i am not trying to use Oauth, i use normal email/password. And its working great, but i am writing a custom route to handle payment and i need to add an auth middleware on the route and be able to get the user ID, so i can continue the processing.
Thanks for your reply.

**Reply by ganigeorgiev** [reply]
Sorry, I misunderstand the question.

You can make use of the `apis.RequireRecordAuth()` middleware.
You can access the currently authenticated user by reading it from the request context, aka. `c.Get(apis.ContextAuthRecordKey).(*models.Record)`. You can find some more examples in https://pocketbase.io/docs/router/#get-logged-admin-or-app-user-in-a-route-handler.

**Reply by Johny-rescaf** [reply]
Thanks alote for the clear answer.

**Reply by Johny-rescaf** [reply]
Thanks alote for the clear answer.
